go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -0,0 +1,94 @@
+package udn
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type Allocator struct {
+	watchFactory factory.NodeWatchFactory
+	nodeName     string
+}
+
+func New(watchFactory factory.NodeWatchFactory, nodeName string) *Allocator {
+	return &Allocator{
+		watchFactory: watchFactory,
+		nodeName:     nodeName,
+	}
+}
+
+func (a *Allocator) AllocateConntrackMark(networkName string) (uint, error) {
+	return a.calculateIDsFromNetwork("ct-mark", networkName, config.Gateway.UserDefinedNetworkConntrackMarkBase, config.Default.MaxUserDefinedNetworks, 1)
+}
+
+func (a *Allocator) AllocateVRFTable(networkName string) (uint, error) {
```

Comment:
I guess we can use `uint32` for vrf table id consider its max value (2^32-1)

Reply:
Whell the `uint` is "just" the type, we are already limiting it with the `MaxUserDefinedNetworks` that by default is 30k I doubt we are going to allow configure more network than that.

Reply:
30,000 is itself high.